src/main/flight/pid.c

```diff
@@ -854,23 +862,30 @@ void FAST_CODE pidController(const pidProfile_t *pidProfile, timeUs_t currentTim
         } else {
             levelMode = LEVEL_MODE_RP;
         }
-    } else {
-        levelMode = LEVEL_MODE_OFF;
-    }
 
-    // Keep track of when we entered a self-level mode so that we can
-    // add a guard time before crash recovery can activate.
-    // Also reset the guard time whenever GPS Rescue is activated.
-    if (levelMode) {
+        // Keep track of when we entered a self-level mode so that we can
+        // add a guard time before crash recovery can activate.
+        // Also reset the guard time whenever GPS Rescue is activated.
         if ((levelModeStartTimeUs == 0) || (gpsRescueIsActive && !gpsRescuePreviousState)) {
             levelModeStartTimeUs = currentTimeUs;
         }
+
+        // Calc horizonLevelStrength if needed
+        if (FLIGHT_MODE(HORIZON_MODE)) {
+            horizonLevelStrength = calcHorizonLevelStrength();
```

Comment:
You're now calling `calcHorizonLevelStrength()` from a routine declared as FAST_CODE whereas previously it was being called from `pidLevel()` which is declared as `FAST_CODE_NOINLINE`. This results in an overflow of `ITCM_RAM` on F7 builds.

I therefore suggest the application of the following patch to this PR.
```
diff --git a/src/main/flight/pid.c b/src/main/flight/pid.c
index 481b6c9..629b3ef 100644
--- a/src/main/flight/pid.c
+++ b/src/main/flight/pid.c
@@ -369,7 +369,7 @@ static float getLevelModeRcDeflection(uint8_t axis)
 }
 
 // calculates strength of horizon leveling; 0 = none, 1.0 = most leveling
-STATIC_UNIT_TESTED float calcHorizonLevelStrength()
+STATIC_UNIT_TESTED FAST_CODE_NOINLINE float calcHorizonLevelStrength()
 {
     // start with 1.0 at center stick, 0.0 at max stick deflection:
     float horizonLevelStrength = 1.0f - MAX(fabsf(getLevelModeRcDeflection(FD_ROLL)), fabsf(getLevelModeRcDeflection(FD_PITCH)));
```

Reply:
Makes sense. I will do it in the evening. Thank you.